src/chainparams.cpp

```diff
@@ -271,6 +271,10 @@ class CMainParams : public CChainParams {
         consensus.nMTPSwitchTime = SWITCH_TO_MTP_BLOCK_HEADER;
         consensus.nMTPStartBlock = 117564;
         consensus.nMTPFiveMinutesStartBlock = SWITCH_TO_MTP_5MIN_BLOCK;
+
+        consensus.nPPSwitchTime = SWITCH_PROGPOW_BLOCK_HEADER;
+        assert(consensus.nMTPSwitchTime < consensus.nPPSwitchTime && "PP Switch time must be after MTP");
+
```

Comment:
there should be testnet parameters as well. And some way to test it on regtest

Reply:
Need to discuss this about how to implement and what support we have from someone willing to set up a testnet pool.

Reply:
Mintpond or 2miners may be able to assist here.

Reply:
Don't forget unit tests. For this you'll need a way to set up HF block time dynamically every time the test is run